Library/Homebrew/keg_relocate.rb

```diff
@@ -165,6 +165,49 @@ def replace_text_in_files(relocation, files: nil)
     changed_files
   end
 
+  def relocate_build_prefix(keg, old_prefix, new_prefix)
+    # Find binaries which match prefix strings.
+    string_matches = Set.new
+    keg.each_unique_file_matching(old_prefix) do |file|
+      string_matches << file
+    end
+
+    binary_string_matches = Set.new
+    keg.each_unique_binary_file do |file|
+      binary_string_matches << file if string_matches.include?(file)
+    end
+
+    # Only consider string matches which are binary files with null bytes, and remove any
+    # matches which are sharballs found by text_files.
+    binary_string_matches -= text_files
+
+    # Split binary by null characters into array and substitute new cellar for old cellar.
+    # Null padding is added if the new string is too short.
+    binary_string_matches.each do |binary_file|
+      binary_file.ensure_writable do
+        binary = File.binread binary_file
+        puts "Replacing build prefix in: #{binary_file}"
```

Comment:
This should probably be `odebug` or require `--verbose` or `HOMEBREW_DEVELOPER`.

Reply:
I gated this behind `verbose` as I do think it's useful in kegs with a lot of files that need patching to be able to see this progress and know that replacement is happening successfully, but should not displayed by default.